vars/sfdxBuildPipeline.groovy

```diff
@@ -150,6 +180,33 @@ def call(Map parameters = [:]) {
             stage("publish") {
                 junit keepLongStdio: true, testResults: 'tests/**/*-junit.xml'
             }
+
+            if (notificationChannel?.trim()) {
+                stage("slack notification end") {
+                    /*
+                    * Slack integration
+                    * https://api.slack.com/docs/message-guidelines
+                    */
+                    def slackNotificationColor;
+                    if ("${currentBuild.currentResult}" == "UNSTABLE") {
+                        slackNotificationColor = 'warning'
+                    } else if ("${currentBuild.currentResult}" == "FAILURE") {
+                        slackNotificationColor = 'danger'
+                    } else if ("${currentBuild.currentResult}" == "SUCCESS") {
+                        slackNotificationColor = 'good'
+                    }
+                    try {
+                        slackSend(
+                            channel: "${notificationChannel}",
+                            color: slackNotificationColor,
+                            message: "${URLDecoder.decode(env.JOB_NAME)} - #${env.BUILD_NUMBER} - ${currentBuild.currentResult} after ${currentBuild.durationString.minus(' and counting')} (<${env.BUILD_URL}|Open>)"
```

Comment:
Is there a reason to perform `URLDecoder.decode` here and not when the build started?

Reply:
Yes, there is. If your URL has %20, or %252, the we don't want to send that part of the message.

Reply:
Fair enough, I guess it means you need to the `decode` on line 64 then. Potentially keeping it a variable to avoid duplication.